# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/BrowserProcessSubThread.cpp
# Commit: 48088463c656
# Full Hash: 48088463c6568b03a3e6a4ca6417dd9b650ced2a
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-22 21:29:07
# Regressor Bug: 1706374
# File Overlap Count: 2
# Description:
#   Bug 1706374 - Part 12a: Initialize NodeController when creating IO thread, r=handyman
#   
#   This also consumes the existing channel created when launching a process to
#   create the the conneciton required by NodeController for communicating between
#   processes. In part 12b, consumers of the broken APIs will be adjusted to use
# ==============================================================================

diff -r b09ae4c3a94b -r 48088463c656 ipc/glue/BrowserProcessSubThread.cpp
--- a/ipc/glue/BrowserProcessSubThread.cpp	Mon Jun 21 21:53:10 2021 +0000
+++ b/ipc/glue/BrowserProcessSubThread.cpp	Mon Jun 21 21:53:10 2021 +0000
@@ -5,6 +5,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "mozilla/ipc/BrowserProcessSubThread.h"
+#include "mozilla/ipc/NodeController.h"
 
 #if defined(OS_WIN)
 #  include <objbase.h>
@@ -49,9 +50,18 @@
   // Initializes the COM library on the current thread.
   CoInitialize(nullptr);
 #endif
+
+  // Initialize the ports library in the current thread.
+  if (mIdentifier == IO) {
+    NodeController::InitBrokerProcess();
+  }
 }
 
 void BrowserProcessSubThread::CleanUp() {
+  if (mIdentifier == IO) {
+    NodeController::CleanUp();
+  }
+
 #if defined(OS_WIN)
   // Closes the COM library on the current thread. CoInitialize must
   // be balanced by a corresponding call to CoUninitialize.